packages/form/templates/solid-js/Form.tsx

```diff
@@ -0,0 +1,135 @@
+// Generated by simple:form
+
+import {
+	type ComponentProps,
+	For,
+	Show,
+	createContext,
+	createSignal,
+	useContext,
+} from "solid-js";
+import {
+	type FieldErrors,
+	type FormState,
+	type FormValidator,
+	formNameInputProps,
+	getInitialFormState,
+	toSetValidationErrors,
+	toTrackAstroSubmitStatus,
+	toValidateField,
+	validateForm,
+} from "simple:form";
+
+export function useCreateFormContext(
+	validator: FormValidator,
+	fieldErrors?: FieldErrors,
+) {
+	const initial = getInitialFormState({ validator, fieldErrors });
+	const [formState, setFormState] = createSignal<FormState>(initial);
+	return {
+		value: formState,
+		set: setFormState,
+		setValidationErrors: toSetValidationErrors(setFormState),
+		validateField: toValidateField(setFormState),
+		trackAstroSubmitStatus: toTrackAstroSubmitStatus(setFormState),
+	};
+}
+
+export function useFormContext() {
+	const formContext = useContext(FormContext);
+	if (!formContext) {
+		throw new Error(
+			"Form context not found. `useFormContext()` should only be called from children of a <Form> component.",
+		);
+	}
+	return formContext;
+}
+
+type FormContextType = ReturnType<typeof useCreateFormContext>;
+
+const FormContext = createContext<FormContextType | undefined>(undefined);
+
+export function Form(
+	props: {
+		validator: FormValidator;
+		context?: FormContextType;
+		fieldErrors?: FieldErrors;
+	} & Omit<ComponentProps<"form">, "method" | "onSubmit">,
+) {
+	const formContext =
+		props.context ?? useCreateFormContext(props.validator, props.fieldErrors);
+
+	return (
+		<FormContext.Provider value={formContext}>
+			<form
+				{...props}
+				method="post"
+				onSubmit={async (e) => {
+					const formData = new FormData(e.currentTarget);
+					formContext.set((formState) => ({
+						...formState,
+						isSubmitPending: true,
+						submitStatus: "validating",
+					}));
+					const parsed = await validateForm({
+						formData,
+						validator: props.validator,
+					});
+					if (parsed.data) {
+						return formContext.trackAstroSubmitStatus();
+					}
+
+					e.preventDefault();
+					e.stopPropagation();
+					formContext.setValidationErrors(parsed.fieldErrors);
+				}}
+			>
+				<Show when={props.name}>
+					{(name) => <input {...formNameInputProps} value={name()} />}
+				</Show>
+				{props.children}
+			</form>
+		</FormContext.Provider>
+	);
+}
+
+export function Input(inputProps: ComponentProps<"input"> & { name: string }) {
+	const formContext = useFormContext();
+	const fieldState = () => formContext.value().fields[inputProps.name];
+	if (!fieldState()) {
+		throw new Error(
+			`Input "${inputProps.name}" not found in form. Did you use the <Form> component?`,
+		);
+	}
+
+	return (
+		<Show when={fieldState()}>
```

Comment:
I agree but if I do this, it says it can be undefined 😭

![image](https://github.com/bholmesdev/simple-stack/assets/13603345/3a3e7ada-54d9-4ace-8571-feb45a161114)

Should I add the `!`?

Reply:
And if I do this, it loses reactivity

![image](https://github.com/bholmesdev/simple-stack/assets/13603345/349a1b93-ca82-4be2-a460-e483ceadca03)

Reply:
Do you have other ideas? I could change the tsconfig for this template to strict instead of strictest and it will work in [this approach](https://github.com/bholmesdev/simple-stack/pull/24#discussion_r1440468414)

Reply:
Ahh thanks, I see the issue now. I'd like to keep `strictest` since this will be generated code in the user's project, and I want to pass the strictest config possible to ensure we don't raise type errors in your project.

I figured out the problem though! We need to hoist the state _inside_ the state function. That way it can guarantee the function never returns `undefined`, since it's always mapped to an error. The solution you showed above does not guarantee that. You can check `fieldState() !== undefined`, but what if calling that function a second time returns a different value? TypeScript unfortunately isn't smart enough to see that.
<img width="599" alt="image" src="https://github.com/bholmesdev/simple-stack/assets/51384119/a1e3e623-84a4-4d73-8883-5c8a26e6ab29">